zerostash/src/commands/zfs/zfs_commands.rs

```diff
@@ -0,0 +1,33 @@
+use std::{boxed::Box, future::Future, marker::Send, pin::Pin};
+
+use clap::Parser;
+
+use crate::prelude::AsyncRunnable;
+
+use super::{ZfsCommit, ZfsDestroy, ZfsExtract, ZfsLs};
+
+/// Zerostash Subcommands
+/// Subcommands need to be listed in an enum.
+#[derive(Debug, Parser)]
+pub enum ZerostashZfs {
+    /// Add a ZFS snapshot to the stash
+    Commit(ZfsCommit),
+
+    /// Extracts a snapshot to stdout
+    Extract(ZfsExtract),
+
+    /// Remove a snapshot from the stash
+    Destroy(ZfsDestroy),
+
+    /// List Snapshots in a stash
+    Ls(ZfsLs),
+}
+
+pub fn match_zfs_cmd<'a>(cmd: &'a ZerostashZfs) -> Pin<Box<dyn Future<Output = ()> + Send + 'a>> {
```

Comment:
I think you should be able to implement `AsyncRunnable` here, just like how the other commands do it. Take a look at `keys.rs` which does this ad absurdum.